### PR TITLE
FIREFLY-1376: Convert textarea to JoyUI

### PR DESCRIPTION
**Pull request by jaladh-singhal**
Fixes [FIREFLY-1376](https://jira.ipac.caltech.edu/browse/FIREFLY-1376)

- Converted InputAreaFieldView to use joyui Textarea
  - exposed common code in InputFieldView to use here
  - cleaned up lot of styling props and css file no longer needed
- Updated usages of InputAreaFieldView (see testing), cleaning up old props

Note: joyui Textarea uses minRows and maxRows to control number of lines shown in textarea initially and to control maximum lines it can expand to. This means setting minRows=maxRows will create a fixed height textarea which is what old`rows` prop was controlling.

### Extras:
- [x] improve spacing and layout of CatalogSearch panel
- [x] add spacing guidelines to documentation

## Testing
https://fireflydev.ipac.caltech.edu/firefly-1376-textarea/firefly/

Check following:
- Catalog search panel 
   - change search method to polygon -> coordinates **textarea**
   - additional SQL constraints **textarea**
- TriViewPanel
   - table options -> advanced filters tab -> constraints **textarea**
   - chart filters -> filters **textarea**
   - select an area in coverage -> binoculors btn -> search refinement option -> polygon radio btn -> coordinates **textarea** (yellow colored because it's connected)
- TAP search -> Edit ADQL -> ADQL **textarea** (with prismjs coloring - background as well as text)

Extras:
Visually inspect the catalog selection panel (or by web inspector), there should be no unnecessary paddings. And layout shouldn't jump vertically when changing catalog or search method.

### CODE REVIEW

File: src/firefly/js/ui/ThemeSetup.js [hunk 1]
```diff
@@ -89,18 +89,6 @@ export function defaultTheme() {
 
 
 /*
- *  UI Notes
- * - toolbar background: <Sheet variant='soft'> or <Sheet>
-        <Sheet className='TapSearch__section' variant='outline' sx={{flexDirection: 'column', flexGrow: 1}}>
+ *  UI Notes: refer to /docs/ui-notes.md and add stuff there
```
Comment (robyww) [inline src/firefly/js/ui/ThemeSetup.js:92]: Thanks are starting this document

File: docs/ui-notes.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+### Buttons
+- primary button:  `<Button size:'md', variant: 'solid' color: 'primary'/>`
+- secondary button:  `<Button/>` (see above for default)
+- significant action button: (eg file upload): `<Button color='success' variant='solid'/>`
+- other buttons: `<Chip/>` (see above for defaults)
+
+### Text
+- information: `<Typography/>`
+- labels: `<Typography/>`
+- feedback: `<Typography color='warning'/>`
```
Comment (loitly) [inline docs/ui-notes.md:15]: Good point.  I wonder if we should further define level as well. e.g.
```js
label: <Typography level='title-x'/>
information: <Typography level='body-x'/>
```

**Review by robyww — APPROVED** [review]
This all looks good. Thanks for creating the document and the guidelines on layout.

File: docs/ui-notes.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# UI Notes
+
+### Miscellaneous
+- toolbar background: `<Sheet variant='soft'>` or `<Sheet className='TapSearch__section' variant='outline' sx={{flexDirection: 'column', flexGrow: 1}}>`
+
```
Comment (loitly) [inline docs/ui-notes.md:5]: Not sure about 'TapSearch__section'.  I feel that should be replaced with Sheet or Card.

File: src/firefly/js/ui/CatalogSearchMethodType.jsx [hunk 4]
```diff
@@ -286,10 +282,11 @@ function SizeArea({groupKey, searchType, imageCornerCalc}) {
         return radiusInField() ;
     } else if (searchType === SpatialMethod.Elliptical.value) {
         return (
-            <Stack {...{p:.5, flexWrap:'no-wrap', spacing:1 }}>
+            <Stack {...{spacing: 1 }}>
```
Comment (loitly) [inline src/firefly/js/ui/CatalogSearchMethodType.jsx:285]: Only spacing is left.  No need to spread.

File: src/firefly/js/ui/CatalogSearchMethodType.jsx [hunk 5]
```diff
@@ -410,12 +401,10 @@ function renderTargetPanel(groupKey, searchType) {
                      searchType === SpatialMethod.Box.value ||
                      searchType === SpatialMethod.Elliptical.value);
     return (
-        <div className='intarget'>
+        <Box sx={{height: 80}}>
```
Comment (loitly) [inline src/firefly/js/ui/CatalogSearchMethodType.jsx:404]: FYI, height is a property of Box.  You can use it as
```js
<Box height={80}>
```
https://mui.com/system/properties/

File: src/firefly/js/visualize/ui/CatalogConstraintsPanel.jsx [hunk 6]
```diff
@@ -130,20 +130,18 @@ export class CatalogConstraintsPanel extends PureComponent {
 
         if (isEmpty(tableModel) || !tableModel.tbl_id.startsWith(catname)) {
             //return <div style={{top: 0}} className='loading-mask'></div>;
-            return <div/>;
+            return <Box sx={{minHeight: '360px'}}/>; //to prevent the outer layout from shrinking while the table loads
```
Comment (loitly) [inline src/firefly/js/visualize/ui/CatalogConstraintsPanel.jsx:133]: FYI, `minHeight: 360` will do as well, px is implied.

**Review by loitly — APPROVED** [review]
Good work on the Input changes.  Everything looks good.

File: docs/ui-notes.md (hunk 2 above)
Comment (jaladh-singhal) [reply docs/ui-notes.md:15]: I just copy pasted it from what was present at ThemeSetup. Also I don't see any level named like these, did you mean 'title-sm', 'body-xs'?

Please feel free to edit it in your upcoming PRs

File: src/firefly/js/ui/ThemeSetup.js (hunk 1 above)
Comment (jaladh-singhal) [reply src/firefly/js/ui/ThemeSetup.js:92]: No problem!

File: docs/ui-notes.md (hunk 3 above)
Comment (jaladh-singhal) [reply docs/ui-notes.md:5]: I agree - I also didn't like that classname is being used. Again feel free to edit it.

File: src/firefly/js/ui/CatalogSearchMethodType.jsx (hunk 5 above)
Comment (jaladh-singhal) [reply src/firefly/js/ui/CatalogSearchMethodType.jsx:404]: Oh ok - thanks for letting me know

File: docs/ui-notes.md (hunk 2 above)
Comment (loitly) [reply docs/ui-notes.md:15]: I mean when text is `body-sm`, then label for that text should be `title-sm`.
Basically, use `title` for label, and `body` for normal text.  They will produce consistent padding.

Comment (jaladh-singhal) [reply docs/ui-notes.md:15]: Oh I see. Yes, that will be helpful!

**Pull request merged**